.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,87 @@
+name: "Setup Virtual Env"
+
+inputs:
+  use_cached_venv:
+    description: 'Use Cached Virtual Env'
+    default: 'true'
+
+runs:
+  using: composite
+  steps:
+    - name: Restore pre-commit cache
+      id: cache-pre-commit
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pre-commit
+        key: v1-pre-commit-${{ env.pythonLocation }}-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    - name: Install pre-commit
+      run: |
+        pip install pre-commit
+        pre-commit install-hooks
+      shell: bash -ileo pipefail {0}
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: '.nvmrc'
+        cache: 'yarn'
+        cache-dependency-path: '**/yarn.lock'
+    - name: Initialize React
+      run: make react-init
+      shell: bash -ileo pipefail {0}
+    - name: Install Dependencies
+      run: |
+        sudo apt update
+        # protobuf, pyodbc, dot, & graphviz dependencies
+        sudo apt install -y gnupg \
+            unixodbc-dev \
+            graphviz \
+            libgvc6
+
+        echo "deb http://ppa.launchpad.net/maarten-fonville/protobuf/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/protobuf.list
+        sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4DEA8909DC6A13A3
+        sudo apt update
+        sudo apt install -y protobuf-compiler
```

Comment:
Whoops, missed this the first time around, but you can reorder this to

```bash
echo "deb http://ppa.launchpad.net/maarten-fonville/protobuf/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/protobuf.list
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4DEA8909DC6A13A3
sudo apt update
sudo apt install -y gnupg \
    unixodbc-dev \
    graphviz \
    libgvc6 \
    protobuf-compiler
```

to avoid having to run `apt update` multiple times